packages/react-storage/src/components/StorageImage/StorageImage.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+
+import { Image, ComponentClassNames } from '@aws-amplify/ui-react';
+import { useStorageURL } from '@aws-amplify/ui-react/internal';
+
+import type { StorageImageProps } from './types';
+
+export const StorageImage = ({
+  imgKey,
+  accessLevel,
+  identityId,
+  fallbackSrc,
+  onStorageError,
+  ...rest
+}: StorageImageProps): JSX.Element => {
+  const { url, error } = useStorageURL(
```

Comment:
Can we prevent this hook from running if the `src` prop is provided?

Reply:
Think we decided to omit the `src` prop on `StorageImage` and instead provide  a `fallbackSrc` prop which makes more sense semantically and functionally in case the Storage.get call gets rejected. People should go right ahead using the `Image` primitive with `src` prop.